src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(String operation, String fruit, String quantity) {
+        this.operation = Operation.of(operation);
+        this.fruit = fruit;
+        this.quantity = Integer.parseInt(quantity);
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
```

Comment:
The constructor for the enum Operation should be private since enums define a finite set of values and should not allow instantiation outside of the enum itself.

Reply:
In the Java programming language, the constructor for an enum cannot be invoked directly. Hence, the constructors in an enum type are always private or package-private. They are private by default if you do not specify an access modifier.
Therefore, it's not necessary to declare the constructor as private, although it's a common practice to do so for clarity and consistency with other classes.